pkg/util/json/json.go

```diff
@@ -89,6 +89,10 @@ type JSON interface {
 	// Exists implements the `?` operator.
 	Exists(string) bool
 
+	// ObjectKeyIterator is an iterator to access the keys of an object, and it returns nil if the obj
+	// is not an object.
+	ObjectKeyIterator() <-chan JSON
```

Comment:
Would you mind naming this `IterObjectKeys`? Just because it will match up better with the iterators introduced in #20217.

Also, I think this might be more general and useful if it returned strings rather than JSON? That's fine because the keys are always strings.

Reply:
I think so too. I'll do that!